client/components/purchase-detail/index.jsx

```diff
@@ -70,6 +70,7 @@ PurchaseDetail.propTypes = {
 	description: React.PropTypes.oneOfType( [
 		React.PropTypes.array,
 		React.PropTypes.string,
+		React.PropTypes.object
```

Comment:
Good idea! However, I think only the latter two are returned by `this.translate` - the addition here is for when description is a single `ReactElement`.